codalab/worker/local_run/docker_image_manager.py

```diff
@@ -102,13 +104,20 @@ def _cleanup(self):
                     entry_to_remove.digest,
                 )
                 try:
-                    image_to_delete = self._docker.images.get(entry_to_remove.id)
-                    tags_to_delete = image_to_delete.tags
-                    for tag in tags_to_delete:
-                        self._docker.images.remove(tag)
-                    # if we successfully removed the image also remove its cache entry
+                    # We try to delete images from image cache in LRU order
+                    self._docker.images.remove(entry_to_remove.id)
                     del self._image_cache[entry_to_remove.digest]
-                except docker.errors.NotFound:
+
+                    # We try to remove any dangling docker images
+                    deleted_dangling_entries = self._docker.images.prune(filters={'dangling': True})
```

Comment:
I added this part to remove any dangling images which might be helpful a little bit?

Reply:
I think dangling images are mainly caused by locally building images and not pulling finished images from Dockerhub like we do, so I don't think this will make a big difference.